### ISSUE TITLE
[BUG] git+ssh:// and git+https:// dependencies break npm commands on nodejs >8.16.0

### ISSUE THREAD

**Issue by bobpaul** [npm/cli#1343]
**Steps To Reproduce**
1. `nvm use 8.16.0`

2. `run "mkdir main-project && cd main-project && npm init -y"`

3. edit the package.json (main-project) and add:

```
"dependencies": {
    "git-project": "git+ssh://foo@foo.com/foo.git"
  }
```
or
```
"dependencies": {
    "git-project": "git+https://foo.com/foo.git"
  }
```
or if the repository is a public github repo:
```
"dependencies": {
    "git-project": "userfoo/foo.git"
  }
```
4. `yarn`
5. `npm ls`
this works
6. `nvm use 8.17.0`
7. `npm ls`
results in `npm ERR! Cannot read property '0' of undefined`

**Expected Behavior**
`npm ls` should work on regardless of the selected node version

**Additional Details**
For step 8 I've also tested many versions of node 10.x.x and node 12.x.x. I've not found any that work yet.

For steps 5 and 7 I've also tested `npm prune --production`. In fact, this is where I first noticed the breakage, but `ls` is easier for testing.

I've reproduced this on Linux, Mac, and Windows.

**Comment by isaacs** [comment]
When you set the node version to 8.17.0, you're _also_ setting the npm version to 6.13.4.  In other words, this is a bug that was fixed some time between `6.13.4` and `6.14.5`.  Short of a time machine, there's not much I can do to fix bugs in the past, unfortunately.  But, you can work around this by doing `nvm install-latest-npm`.  (`nvm use 8.17.0 --install-latest-npm` might work also, I'm not sure.)

**Comment by ljharb** [comment]
`nvm install 8.17.0 --latest-npm` will work!
